sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/ReceiverUnsettledDeliveries.java

```diff
@@ -0,0 +1,739 @@
+     */
+    private void handleReleasedOrUnknownRemoteOutcome(DispositionWork work, Delivery delivery, Outcome remoteOutcome) {
+        final AmqpErrorContext amqpErrorContext = getErrorContext(delivery);
+        final AmqpException completionError;
+
+        final DeliveryStateType remoteOutcomeType = delivery.getRemoteState().getType();
+        if (remoteOutcomeType == DeliveryStateType.Released) {
+            completionError = new AmqpException(false, AmqpErrorCondition.OPERATION_CANCELLED,
+                "AMQP layer unexpectedly aborted or disconnected.", amqpErrorContext);
+        } else {
+            completionError = new AmqpException(false, remoteOutcome.toString(), amqpErrorContext);
+        }
+
+        logger.atInfo()
+            .addKeyValue(DELIVERY_TAG_KEY, work.getDeliveryTag())
+            .addKeyValue(DELIVERY_STATE_KEY, delivery.getRemoteState())
+            .log("Completing pending updateState operation with exception.", completionError);
+
+        completeDispositionWorkWithSettle(work, delivery, completionError);
+    }
+
+    /**
+     * Iterate through all the current {@link DispositionWork} and complete the work those are timed out.
+     */
+    private void completeDispositionWorksOnTimeout(String callSite) {
+        if (pendingDispositions.isEmpty()) {
+            return;
+        }
+
+        final int[] completionCount = new int[1];
+        final StringJoiner deliveryTags = new StringJoiner(", ");
+
+        pendingDispositions.forEach((deliveryTag, work) -> {
+            if (work == null || !work.hasTimedout()) {
+                return;
+            }
+
+            if (completionCount[0] == 0) {
+                logger.info("Starting completion of timed out disposition works (call site:{}).", callSite);
+            }
+
+            final Throwable completionError;
+            if (work.getRejectedOutcomeError() != null) {
+                completionError = work.getRejectedOutcomeError();
+            } else {
+                completionError = new AmqpException(true, AmqpErrorCondition.TIMEOUT_ERROR,
+                    "Update disposition request timed out.", getErrorContext(deliveries.get(work.getDeliveryTag())));
+            }
+            deliveryTags.add(work.getDeliveryTag());
+            completeDispositionWork(work, completionError);
+            completionCount[0]++;
+        });
+
+        if (completionCount[0] > 0) {
+            // The log help debug if the user code chained to the work-mono (DispositionWork::getMono()) never returns.
+            logger.info("Completed {} timed-out disposition works (call site:{}). Locks {}",
+                callSite, completionCount[0], deliveryTags.toString());
+        }
+    }
+
+    /**
+     * Iterate through all the {@link DispositionWork}, and 'force' to complete the uncompleted works because
+     * this {@link ReceiverUnsettledDeliveries} is closed.
+     */
+    private void completeUncompletedDispositionWorksOnClose() {
```

Comment:
Oh, I forgot to mention here. Update to 'completeDispositionWorksOnClose'.

Reply:
Ah.. good catch, I’ll rename.